Interactive markers

LindseyWeiskopf opened the pull request:
Summary: Create interactive map feature that highlights listing

I created a marker listener so that when a marker is clicked, the map will zoom to that marker, paginate towards the listing, and highlight the appropriate listing. When the next marker is selected, the past listing should un-highlight 
**Not deployed at the moment

Thank you!

shaquille-hall commented on src/main/webapp/mapScript.js:
```diff
@@ -0,0 +1,44 @@
+    document.getElementById(highlightedId).style.backgroundColor = '#fff';
+    map.setCenter(marker.getPosition());
+    map.setZoom(16);
+    let firstId = document.getElementById('results-content').firstChild.id;
+    // If card is not on the current page
+    if ((cardNumber-firstId < 0) || (cardNumber-firstId >= 3)) {
+      displayCards(cardNumber-firstId);
+    }
```
Curious. Wouldn't firstId always be 0. So unless cardNumber is also referencing the first card wouldn't this first half effectively just be true. Have you considered using a boolean to track the selected card?

LindseyWeiskopf replied:
Since the results-content section only holds three children, the firstId will be the id of whatever card is first on the current page (ex. I scroll to the third page and then click a marker). The first half would be true only if the card is one of the three on the current page that the user is paginated to. Does that make sense? I decided to not use a boolean because only one card should be selected at a time, and ideally that card will only be unhighlighted when another marker is chosen to avoid confusion. An alternative would be to have the chosen card flash when the marker is clicked if you think that is a better solution then carrying over the highlighted card to the next call of the function.

LindseyWeiskopf commented on src/main/webapp/mapScript.js:
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+let highlightedId = 0;
```
Fixed (got rid of the variable all together)

LindseyWeiskopf commented:
I fixed the page augment, so the markers should not get messed up if the user paginates to a different page. I also changed the highlight so that the card flashes for 5 seconds to avoid carrying over a highlightedCard variable for each click of a marker.

shaquille-hall commented on src/main/webapp/mapScript.js:
```diff
@@ -12,18 +12,36 @@
+
+  marker.addListener('click', function() {
+    map.setCenter(marker.getPosition());
+    map.setZoom(18);
```
Confirming this is meant to be different from the original zoom value. 18 != 13

LindseyWeiskopf replied:
Yes -- @AbakirH found a function where you can fit the bounds of the map to the markers so the initial map should be set to the best fit zoom and then it zooms more when a marker is clicked.

shaquille-hall commented on src/main/webapp/mapScript.js:
```diff
@@ -12,18 +12,36 @@
+    if ((cardNumber-firstCardId < 0) || (cardNumber-firstCardId >= 3)) {
+      displayCards(Math.floor((cardNumber-firstCardId)/3)*3);
+    }
+    let currentCard = document.getElementById(cardNumber);
```
nit: Please add a space above this line.

shaquille-hall commented on src/main/webapp/mapScript.js:
```diff
@@ -12,18 +12,36 @@
+      displayCards(Math.floor((cardNumber-firstCardId)/3)*3);
+    }
+    let currentCard = document.getElementById(cardNumber);
+    // Cooresponding card flashes for 5 seconds
```
*Corresponding

LindseyWeiskopf replied:
fixed (spelling is hard!)
